node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -283,37 +281,31 @@ impl Initialized {
 		self.participation.process_active_leaves_update(ctx, &update).await?;
 
 		if let Some(new_leaf) = update.activated {
-			match self
-				.rolling_session_window
-				.cache_session_info_for_head(ctx.sender(), new_leaf.hash)
-				.await
-			{
+			let session_idx =
+				self.runtime_info.get_session_index_for_child(ctx.sender(), new_leaf.hash).await;
+
+			match session_idx {
+				Ok(session_idx) => {
+					// Dummy fetch to ensure the session is cached
+					let _ = self
+						.runtime_info
+						.get_session_info_by_index(ctx.sender(), new_leaf.hash, session_idx)
+						.await?;
+
+					if session_idx > self.highest_session {
+						self.highest_session = session_idx;
+						self.spam_slots
+							.prune_old(session_idx.saturating_sub(SESSION_WINDOW_SIZE.get() - 1));
```

Comment:
This is the simplification I wrote about in the PR description. Is it okay to prune spam slots without considering if the session with the last finalized block falls in `session_idx.saturating_sub(SESSION_WINDOW_SIZE.get() - 1)`?